c2rust-analyze/src/context.rs

```diff
@@ -134,6 +134,17 @@ impl PermissionSet {
     //
     // `.union` is used here since it's a `const fn`, unlike `BitOr::bitor`.
     pub const STRING_LITERAL: Self = Self::READ.union(Self::OFFSET_ADD);
+
+    #[cfg(test)]
+    pub const fn union_all<const N: usize>(a: [Self; N]) -> Self {
+        let mut this = Self::empty();
+        let mut i = 0;
+        while i < N {
+            this = this.union(a[i]);
+            i += 1;
+        }
+        this
```

Comment:
Why not use a for loop here?

Reply:
`for` loops can't be used in `const` `fn`s since they call `Iterator::next`, a `trait` method that can't be `const`.